### ISSUE TITLE
Public API for RepositoryContentDescriptor, Filters and accessing dependencies on repositories

### ISSUE THREAD

**Issue by NikkyAI** [gradle/gradle#13952]
Currently there is no way for plugins to inspect what has been configured in repository content blocks
Or for plugins to use gradle's existing code to do the same requests to the repositories while respecting the filters set in the content blocks
because its all internal / private / package-private classes

### Expected Behavior

public API for
- `repository.listVersions(moduleIdentifier)`
- `repository.get(moduleComponentIdentifier)` // or getPom ?

change to `repository.contentFilter` to get access to the default (empty) filter
maybe `contentFilterOrDefault` ? or `repository.createContentFilterIfNotExists()` ?

### Current Behavior

internal API, private classes, most logic inaccessible for buildscripts or plugins

### Context

this causes https://github.com/jmfayard/refreshVersions/issues/175 
TL;DR refreshVersions needs to be able to list available versions of a dependency in the configured repositories, but currently there is no way to respect the configuration done in content blocks

I have attempted to make do with the internal api.. and though re-implementing `ModuleIdentifier` (because DefaultModuleIdentifier is not accessible) i can at the very least tell if a specific dependency exists in a repository (but only if a content block was defined on it)
this may allow us to patch the issue a little bit and eliminate repositories .. and do the usual thing for the rest.. but it is *not* ideal to work with internal api

**Comment by alllex** [comment]
Sorry for the late reply

The issue is in the backlog of the relevant team and is prioritized by them.
